proposals/mainnet_migration.md

```diff
@@ -0,0 +1,528 @@
+
+##### Assets
+
+From the [registered modules](#modules), the following ones add an entry to `b.assets`:
+
+- Legacy module
+- Token module
+- Auth module
+- DPoS module
+
+How these modules construct their entry for the `assets` property is defined in the next subsections. The verification of their entries is defined in the _Genesis Block Processing_ sections of the respective module LIPs. Note that once all modules add their entries to `b.assets`, this array [must be sorted by increasing value of moduleID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0055.md#validation-1).
+
+In the following, let `accountsState` be the key-value store of the accounts state of `STATE_SNAPSHOT` computed as described [above](#process). That means, for a byte array `addr` representing an address, `accountsState[addr]` is the corresponding account object following the account schema defined in [LIP 0030](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0030.md). Moreover, let `accounts` be the array that contains all values of `accountsState` for which the key is a 20-byte address sorted in lexicographical order of their `address` property. Correspondingly, let `legacyAccounts` be the array that contains all values of `accountsState` for which the key is an 8-byte address sorted in lexicographical order of their `address` property.
+
+###### Assets Entry for Legacy Module
+
+Let `genesisLegacyStoreSchema` be as defined in [LIP 0050](https://github.com/LiskHQ/lips/blob/aebeecc40447e4229f14fc641fb11527270dfbb5/proposals/lip-0050.md#genesis-state-initialization). The `assets` entry for the legacy module is added by the logic defined in the function `addLegacyModuleEntry` in the following pseudo code:
+
+```python
+addLegacyModuleEntry():
+    legacyObj = {}
+    legacyObj.accounts = []
+    for every account in legacyAccounts:
+        userObj = {}
+        userObj.address = account.address
+        userObj.balance = account.token.balance
+        legacyObj.accounts.append(userObj)
+    data = serialization of legacyObj using genesisLegacyStoreSchema
+    append {"moduleID": MODULE_ID_LEGACY, "data": data} to b.assets
+```
+
+###### Assets Entry for Token Module
+
+Let `genesisTokenStoreSchema` be as defined in [LIP 0051](https://github.com/LiskHQ/lips/blob/aebeecc40447e4229f14fc641fb11527270dfbb5/proposals/lip-0051.md#genesis-assets-schema). The `assets` entry for the token module is added by the logic defined in the function `addTokenModuleEntry` in the following pseudo code:
+
+```python
+addTokenModuleEntry():
+    tokenObj = {}
+    tokenObj.userSubstore = createUserSubstoreArray()
+    tokenObj.supplySubstore = createSupplySubstoreArray()
+    tokenObj.escrowSubstore = []
+    tokenObj.availableLocalIDSubstore = {}
+    tokenObj.availableLocalIDSubstore.nextAvailableLocalID = LOCAL_ID_LSK + 1
+    tokenObj.terminatedEscrowSubstore = []
```

Comment:
I am not sure this substore still exists.

Reply:
Removed.